### ISSUE TITLE
Properly assessing our methods relative to Alter et al

### ISSUE THREAD

**Issue by aarmey** [meyer-lab/systemsSerology#78]
https://github.com/meyer-lab/systemsSerology/blob/e8a4abaeeb132c82e8f6950c5433bdf29016df4f/syserol/regression.py#L42

I'm not sure that we're correctly assessing our predictions. We should be building the elastic net model with all of the patients, but then only assessing the accuracy with the same patients from Alter when comparing to their results.

When we evaluate our ability to predict the patients with missing values, we should again build the model with all the patients, then only look at the results for the patients missing from Alter.

In other words, we should be subsetting the patients only after model building and cross-validation, not before.

I think something that might help is better organizing this code. Why not make a function that gives you the patient indices corresponding to Alter? Then you won't need all this data frame manipulation in multiple places. You can then also have one function to apply the regression model, and just subset the results from that.

**Comment by murphymadeleine21** [comment]
Okay, this sounds good and I will start implementing this, but I do have one question. One of the reasons we have been subsetting it before is because the regression cannot take NaN values. For example, `noCMTF_function_prediction` technically does try to use all of the subjects, but after we run it we are only left with 174/181 subjects. So the indexing/ordering gets out of place. I suppose we may just have to also track which indices are dropped because they have NaN values?

**Comment by aarmey** [comment]
Our methods don’t operate on the original data—they use the decomposition, which doesn’t have missing values.

**Comment by murphymadeleine21** [comment]
Ah yes, that's true. But when we go to run the regression on the decomposed data, we have to align it with the functional Y data, causing us to drop values because some of the functional data are missing.

**Comment by murphymadeleine21** [comment]
Also, I'm re-evaluating how we built the "Alter DataFrame". I'm not entirely convinced that we are supposed to be including the functional data in the original Alter DF? I'm looking through their code. 
`# build models to predict from each feature set, to each function and subject classification
featuresets <- list(luminex=luminex.features, igg=igg.features, glycan=glycan.features, 
                    luminex.gsubjs=luminex.features[rownames(glycan.features),],
                    luminex.plus.glycan=cbind(luminex.features[rownames(glycan.features),], glycan.features))
modelsets.4 <- build.modelsets(featuresets, functions, subject.classes, alpha=0.4)`

It's a bit confusing, but it seems like `featuresets` is their DF, and it doesn't include functional data. Then they run `build.modelsets` with `featuresets` as X, and `functions` as Y

**Comment by aarmey** [comment]
Oh, right, if a function's measurement is missing, then it can't be used for training.

**Comment by aarmey** [comment]
Right. The functional measurements are never used in the training data.

**Comment by murphymadeleine21** [comment]
In that case, I think `importAlterDF` is incorrect, because we call `importFunction`
